Modules/posixmodule.c

```diff
@@ -8656,22 +8674,28 @@ os_ptsname_impl(PyObject *module, int fd)
     int ret;
     char name[MAXPATHLEN+1];
 
-    ret = ptsname_r(fd, name, sizeof(name));
+    if (HAVE_PTSNAME_R_RUNTIME) {
+        ret = ptsname_r(fd, name, sizeof(name));
+    }
+    else {
+#if defined(HAVE_PTSNAME)
+        // fallback to `ptsname` if `ptsname_r` is not available in runtime.
+        return from_ptsname(fd);
+#else
+        // unknown error:
+        // both ptsname_r and ptsname are not available for some reason.
+        ret = -1;
```

Comment:
This is a "cannot happen" case: *ptsname* is available on all versions of macOS where configure detects *ptsname_r*.

I'd either drop drop the guard for the block above or replace this bit by an ``#error``.

(And regardless ``ret = -1`` is wrong, the value should be an errno value such as ``ENOSYS``).

Reply:
Done, thanks for the review! 👍

Reply:
> This is a "cannot happen" case: ptsname is available on all versions of macOS where configure detects ptsname_r.

Is there a macOS or iOS platform where ptsname() and ptsname_r() are not available?

Reply:
> Is there a macOS or iOS platform where ptsname() and ptsname_r() are not available?
In short, no.

ptsname_r() is available on macOS 10.13.4 and iOS 11.3 (and later), ptsname() has been available from the start. Or at least, there are no availability annotations in Apple's headers which is a good indication that the API was available from the start.